File: src/main/java/xyz/lncvrt/galaxyboxpvp/events/PlayerMoveListener.java
package xyz.lncvrt.galaxyboxpvp.events;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

public class PlayerMoveListener implements Listener {
    @EventHandler
    private void onSignPlace(PlayerMoveEvent event) {
        Player player = event.getPlayer();
        int maxY = player.getWorld().getMaxHeight();
        if (player.getLocation().getY() >= maxY) {
            Location newLocation = player.getLocation().clone();
            newLocation.setY(maxY - 1);
            player.teleport(newLocation);
        }
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/PlayerMoveListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class PlayerMoveListener : Listener {
    @EventHandler
    private fun onSignPlace(event: PlayerMoveEvent) {
        val player = event.getPlayer()
        val maxY = player.world.maxHeight
        if (player.location.y >= maxY) {
            val newLocation = player.location.clone()
            newLocation.y = (maxY - 1).toDouble()
            player.teleport(newLocation)
        }
    }
}